lib/events/filelog.go

```diff
@@ -136,12 +154,28 @@ func (l *FileLog) EmitAuditEvent(ctx context.Context, event AuditEvent) error {
 
 // EmitAuditEventLegacy adds a new event to the log. Part of auth.IFileLog interface.
 func (l *FileLog) EmitAuditEventLegacy(event Event, fields EventFields) error {
+	l.rw.RLock()
+	defer l.rw.RUnlock()
+
 	// see if the log needs to be rotated
-	err := l.rotateLog()
-	if err != nil {
-		log.Error(err)
+	if l.mightNeedRotation() {
+		// log might need rotation; switch to write-lock
+		// to avoid rotating during concurrent event emission.
+		l.rw.RUnlock()
+		l.rw.Lock()
```

Comment:
this is racy, but add a comment that `l.rotateLog` will perform the "needs rotation" check again internally